### PR TITLE
Adding users list action

### PR DESCRIPTION
**Pull request by nellshamrell**
Supermarket makes heavy use of oc-id.  It would be very useful to be able to retrieve a full list of Chef Server users and their information from oc-id.  This will help us keep our user data in Supermarket in sync with the user data in Chef Server.  Therefore, here is a pull request which adds this.  I also added in some additional specs around the users#show method as I figured out how it worked.

### CODE REVIEW

File: src/oc-id/config/routes.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@
       resources :users, only: :show
 
       get 'me', to: 'users#me'
+      get 'all_users', to: 'users#all_users'
```
Comment (smith) [inline src/oc-id/config/routes.rb]: Is there any reason why we're using this instead of the `index` action of the users controller?

**Comment by mmzyk** [comment]
Doorkeeper is gating this so the only way to access this is when authenticated? I want to make sure we're not opening up a potential security hole where we leak info from the Chef server.

**Comment by mmzyk** [comment]
My other concern is load on the chef server, particularly in an environment like Hosted that will have a lot of users (not sure how often this will be called), but it appears the query the server uses is a single query that if we have the proper indexes on should be fine.

https://github.com/chef/chef-server/blob/master/src%2Foc_erchef%2Fapps%2Foc_chef_wm%2Fsrc%2Foc_chef_wm_users.erl#L124
and
https://github.com/chef/chef-server/blob/master/src%2Foc_erchef%2Fapps%2Fchef_db%2Fpriv%2Fpgsql_statements.config#L522

However, I don't know how long the query will take, as I have no clue if we ever run this in hosted now. Be aware that the possibility of hitting a timeout exists, although I think it should be okay.

**Comment by smith** [comment]
@mm I'm not too concerned about the Chef server load. We'll be hitting the endpoint once every time the cron job runs, which I'm assuming will not be enough to make a difference. If there are missing indexes we can fix them on the server without too much trouble.

I'm not sure this is the right way to do the authorization though. From reading this it seems if I have the app ID of any registered app I can get the whole list of users and whatever comes back from this request. I know we want to make it so users can create applications, so wouldn't this mean that anybody could get that list?

Are we currently doing any privileged API calls through oc-id that a normal user with a normal key couldn't do through `knife raw` or other normal API requests? We should be sure we have it right before unintentionally opening up an information disclosure hole.

**Comment by raskchanky** [comment]
@smith Based on my understanding of the Doorkeeper docs, your assumption sounds correct.  As things stand right now, if I have the app ID of any app, I can get a full list of users for that app, without being authenticated.

It seems like we want to verify that 1) the request coming in for all users is coming from an authenticated user and 2) the authenticated user in question is authorized to view the app that the ID represents.  I'm not sure if `doorkeeper_for` handles both of those cases or not.  I'll research it a bit more.

**Comment by nellshamrell** [comment]
Hi @raskchanky - any thoughts on using doorkeeper_for?

**Comment by nellshamrell** [comment]
I just changed the controller so that doorkeeper_for is required for the all_users action.

**Comment by marcparadise** [comment]
Are we still looking to land this PR?

**Comment by stevendanna** [comment]
@nellshamrell @mmzyk @smith I know everyone is busy, but if anyone knows the status of this work, it would be great if you could post a note here.  Otherwise I'm inclined to close this PR since it hasn't moved in months.

**Pull request closed without merging**